zipline/data/data_portal.py

```diff
@@ -0,0 +1,1186 @@
+                    sid, self._dividends_dict, "DIVIDENDS"
+                ),
+                data,
+                dts,
+                True
+            )
+
+            data *= price_adj_factor
+
+            # if anything is zero, it's a missing bar, so replace it with NaN.
+            # we only want to do this for non-volume fields, because a missing
+            # volume should be 0.
+            data[data == 0] = np.NaN
+
+        np.around(data, 3, out=data)
+
+    @staticmethod
+    def _find_position_of_minute(minute_dt):
+        """
+        Internal method that returns the position of the given minute in the
+        list of every trading minute since market open on 1/2/2002.
+
+        IMPORTANT: This method assumes every day is 390 minutes long, even
+        early closes.  Our minute bcolz files are generated like this to
+        support fast lookup.
+
+        ex. this method would return 2 for 1/2/2002 9:32 AM Eastern.
+
+        Parameters
+        ----------
+        minute_dt: pd.Timestamp
+            The minute whose position should be calculated.
+
+        Returns
+        -------
+        The position of the given minute in the list of all trading minutes
+        since market open on 1/2/2002.
+        """
+        day = minute_dt.date()
+        day_idx = tradingcalendar.trading_days.searchsorted(day) -\
+            INDEX_OF_FIRST_TRADING_DAY
+
+        day_open = pd.Timestamp(
+            datetime(
+                year=day.year,
+                month=day.month,
+                day=day.day,
+                hour=9,
+                minute=31),
+            tz='US/Eastern').tz_convert('UTC')
+
+        minutes_offset = int((minute_dt - day_open).total_seconds()) / 60
+
+        return int((390 * day_idx) + minutes_offset)
+
+    def _get_daily_window_for_sid(self, sid, field, days_in_window,
+                                  extra_slot=True):
+        """
+        Internal method that gets a window of adjusted daily data for a sid
+        and specified date range.  Used to support the history API method for
+        daily bars.
+
+        Parameters
+        ----------
+        sid : int
+            The sid whose data is desired.
+
+        start_dt: pandas.Timestamp
+            The start of the desired window of data.
+
+        bar_count: int
+            The number of days of data to return.
+
+        field: string
+            The specific field to return.  "open", "high", "close_price", etc.
+
+        extra_slot: boolean
+            Whether to allocate an extra slot in the returned numpy array.
+            This extra slot will hold the data for the last partial day.  It's
+            much better to create it here than to create a copy of the array
+            later just to add a slot.
+
+        Returns
+        -------
+        A numpy array with requested values.  Any missing slots filled with
+        nan.
+
+        """
+        daily_data, daily_attrs = self._open_daily_file()
```

Comment:
Without explicit resampling of the raw data by the user, looks like daily history won't work on minute data.